### ISSUE TITLE
How to ssh from Ubuntu on WSL2 to Hyper-V based Ubuntu?

### ISSUE THREAD

**Issue by ismailhkose** [microsoft/WSL#6529]
I can't ssh to Ubuntu on Hyper-V from Ubuntu on WSL2. I have the default configuration on both sides. Both are in different subnets.
How should I configure WSL2 or Hyper-V VM so that they can be in the same network and I can easily make ssh connection from WSL2 to Hyper-V VM?

**Comment by bleepster** [comment]
Hi, I have the same question as OP. Issue #4150 seems to discuss connecting to a WSL service from the Windows host. It does not say anything about connecting to a Hyper-V VM (like Ubuntu) from inside WSL.

So, is this a current limitation - connecting to a Hyper-V VM from inside WSL is not possible? -Or- is there some documentation that describes how to do this?

**Comment by cubic3d** [comment]
> Hi, I have the same question as OP. Issue #4150 seems to discuss connecting to a WSL service from the Windows host. It does not say anything about connecting to a Hyper-V VM (like Ubuntu) from inside WSL. Perhaps, I've just misunderstood the discussion over at #4150 (just disregard this post if I did 😄).
> 
> So, is this a current limitation - connecting to a Hyper-V VM from inside WSL is not possible? -Or- is there some documentation that describes how to do this?

Looks like it isn't a duplicate of #4150 but of #4288. Basically the current workaround is to enable forwarding on the corresponding vSwitch. See https://github.com/microsoft/WSL/issues/4288#issuecomment-652259640 or https://github.com/microsoft/WSL/issues/4288#issuecomment-778790363

**Comment by bleepster** [comment]
> Looks like it isn't a duplicate of #4150 but of #4288. Basically the current workaround is to enable forwarding on the corresponding vSwitch. See [#4288 (comment)](https://github.com/microsoft/WSL/issues/4288#issuecomment-652259640) or [#4288 (comment)](https://github.com/microsoft/WSL/issues/4288#issuecomment-778790363)

Thanks for sharing these! I'll try them out.

----

I ended up with the following after I posted my [question](https://github.com/microsoft/WSL/issues/6529#issuecomment-1008670374):
1) add new interface in my VM (connected to the WSL vSwitch)
2) statically assign an IP address in my VM - based on what is currently assigned to WSL

With this solution, I have to manually change the IP in my VM every time a new IP is assigned to WSL. A little bit of work everytime I use my VM, but it gets the job done.
